chains/evm/deployment/v1_6_4/changesets/configure_allowed_callers.go

```diff
@@ -34,16 +39,25 @@ func configureAllowedCallersApply(mcmsRegistry *changesets.MCMSReaderRegistry) f
 		reports := make([]cldf_ops.Report[any, any], 0)
 
 		// Build the Address and AllowedCallersByChain maps from per-chain inputs
-		addressByChain := make(map[uint64]common.Address)
+		addressesByChain := make(map[uint64]common.Address)
 		allowedCallersByChain := make(map[uint64][]erc20_lock_box.AllowedCallerConfigArgs)
 		for _, perChainInput := range input.ChainInputs {
-			addressByChain[perChainInput.ChainSelector] = perChainInput.Address
+
+			erc20_lock_box_address, err := datastore_utils.FindAndFormatRef(e.DataStore, datastore.AddressRef{
```

Comment:
For this one, you need to allow for the ability to pass in a full ref. Or at least a qualifier string (qualifier is another address ref field). Only because there will be multiple lock boxes for different tokens in the datastore. Qualifier would typically be the token symbol, ultimately depends on what you save to the datastore post-deployment.

Reply:
Qualifier is a good idea. I think the better way would be to use "IOwnable" as the qualifier since the main identifying feature of the lockbox is that it uses Ownable token pools. That way in the future if we build a lockbox which works on RBAC-based token pools we can make the qualifier "IAccessControl". Good Call out. In terms of 1.6.4 there is only one version but forward thinking is also useful.